conda/plugins/manager.py

```diff
@@ -117,24 +126,51 @@ def load_entrypoints(self, group: str, name: str | None = None) -> int:
                     )
                     continue
 
-                # only use the canonical name after this point
-                canonical = self.get_canonical_name(plugin)
-
-                # skip plugin if already registered or blocked
-                if self.get_plugin(canonical) or self.is_blocked(canonical):
-                    continue
```

Comment:
@kenodegard - if I restore this block then things work as expected again.

`conda-libmamba-solver` is for some reason loaded twice in the `defaults` Docker image (first and last):

```
EntryPoint(name='conda-libmamba-solver', value='conda_libmamba_solver.plugin', group='conda')
EntryPoint(name='collect-upstream-conda-tests', value='collect_upstream_conda_tests', group='pytest11')
EntryPoint(name='pyflakes', value='pyflakes.api:main', group='console_scripts')
EntryPoint(name='coverage', value='coverage.cmdline:main', group='console_scripts')
EntryPoint(name='coverage-3.9', value='coverage.cmdline:main', group='console_scripts')
EntryPoint(name='coverage3', value='coverage.cmdline:main', group='console_scripts')
EntryPoint(name='rerunfailures', value='pytest_rerunfailures', group='pytest11')
EntryPoint(name='pytest_forked', value='pytest_forked', group='pytest11')
EntryPoint(name='pycodestyle', value='pycodestyle:_main', group='console_scripts')
EntryPoint(name='conda-verify', value='conda_verify.cli:cli', group='console_scripts')
EntryPoint(name='cph', value='conda_package_handling.cli:main', group='console_scripts')
EntryPoint(name='nodeenv', value='nodeenv:main', group='console_scripts')
EntryPoint(name='timeout', value='pytest_timeout', group='pytest11')
EntryPoint(name='anaconda-project', value='anaconda_project.cli:main', group='console_scripts')
EntryPoint(name='alias', value='setuptools.command.alias:alias', group='distutils.commands')
EntryPoint(name='bdist_egg', value='setuptools.command.bdist_egg:bdist_egg', group='distutils.commands')
EntryPoint(name='bdist_rpm', value='setuptools.command.bdist_rpm:bdist_rpm', group='distutils.commands')
EntryPoint(name='build', value='setuptools.command.build:build', group='distutils.commands')
EntryPoint(name='build_clib', value='setuptools.command.build_clib:build_clib', group='distutils.commands')
EntryPoint(name='build_ext', value='setuptools.command.build_ext:build_ext', group='distutils.commands')
EntryPoint(name='build_py', value='setuptools.command.build_py:build_py', group='distutils.commands')
EntryPoint(name='develop', value='setuptools.command.develop:develop', group='distutils.commands')
EntryPoint(name='dist_info', value='setuptools.command.dist_info:dist_info', group='distutils.commands')
EntryPoint(name='easy_install', value='setuptools.command.easy_install:easy_install', group='distutils.commands')
EntryPoint(name='editable_wheel', value='setuptools.command.editable_wheel:editable_wheel', group='distutils.commands')
EntryPoint(name='egg_info', value='setuptools.command.egg_info:egg_info', group='distutils.commands')
EntryPoint(name='install', value='setuptools.command.install:install', group='distutils.commands')
EntryPoint(name='install_egg_info', value='setuptools.command.install_egg_info:install_egg_info', group='distutils.commands')
EntryPoint(name='install_lib', value='setuptools.command.install_lib:install_lib', group='distutils.commands')
EntryPoint(name='install_scripts', value='setuptools.command.install_scripts:install_scripts', group='distutils.commands')
EntryPoint(name='rotate', value='setuptools.command.rotate:rotate', group='distutils.commands')
EntryPoint(name='saveopts', value='setuptools.command.saveopts:saveopts', group='distutils.commands')
EntryPoint(name='sdist', value='setuptools.command.sdist:sdist', group='distutils.commands')
EntryPoint(name='setopt', value='setuptools.command.setopt:setopt', group='distutils.commands')
EntryPoint(name='test', value='setuptools.command.test:test', group='distutils.commands')
EntryPoint(name='upload_docs', value='setuptools.command.upload_docs:upload_docs', group='distutils.commands')
EntryPoint(name='dependency_links', value='setuptools.dist:assert_string_list', group='distutils.setup_keywords')
EntryPoint(name='eager_resources', value='setuptools.dist:assert_string_list', group='distutils.setup_keywords')
EntryPoint(name='entry_points', value='setuptools.dist:check_entry_points', group='distutils.setup_keywords')
EntryPoint(name='exclude_package_data', value='setuptools.dist:check_package_data', group='distutils.setup_keywords')
EntryPoint(name='extras_require', value='setuptools.dist:check_extras', group='distutils.setup_keywords')
EntryPoint(name='include_package_data', value='setuptools.dist:assert_bool', group='distutils.setup_keywords')
EntryPoint(name='install_requires', value='setuptools.dist:check_requirements', group='distutils.setup_keywords')
EntryPoint(name='namespace_packages', value='setuptools.dist:check_nsp', group='distutils.setup_keywords')
EntryPoint(name='package_data', value='setuptools.dist:check_package_data', group='distutils.setup_keywords')
EntryPoint(name='packages', value='setuptools.dist:check_packages', group='distutils.setup_keywords')
EntryPoint(name='python_requires', value='setuptools.dist:check_specifier', group='distutils.setup_keywords')
EntryPoint(name='setup_requires', value='setuptools.dist:check_requirements', group='distutils.setup_keywords')
EntryPoint(name='test_loader', value='setuptools.dist:check_importable', group='distutils.setup_keywords')
EntryPoint(name='test_runner', value='setuptools.dist:check_importable', group='distutils.setup_keywords')
EntryPoint(name='test_suite', value='setuptools.dist:check_test_suite', group='distutils.setup_keywords')
EntryPoint(name='tests_require', value='setuptools.dist:check_requirements', group='distutils.setup_keywords')
EntryPoint(name='use_2to3', value='setuptools.dist:invalid_unless_false', group='distutils.setup_keywords')
EntryPoint(name='zip_safe', value='setuptools.dist:assert_bool', group='distutils.setup_keywords')
EntryPoint(name='PKG-INFO', value='setuptools.command.egg_info:write_pkg_info', group='egg_info.writers')
EntryPoint(name='dependency_links.txt', value='setuptools.command.egg_info:overwrite_arg', group='egg_info.writers')
EntryPoint(name='eager_resources.txt', value='setuptools.command.egg_info:overwrite_arg', group='egg_info.writers')
EntryPoint(name='entry_points.txt', value='setuptools.command.egg_info:write_entries', group='egg_info.writers')
EntryPoint(name='namespace_packages.txt', value='setuptools.command.egg_info:overwrite_arg', group='egg_info.writers')
EntryPoint(name='requires.txt', value='setuptools.command.egg_info:write_requirements', group='egg_info.writers')
EntryPoint(name='top_level.txt', value='setuptools.command.egg_info:write_toplevel_names', group='egg_info.writers')
EntryPoint(name='keywords', value='setuptools.dist:Distribution._finalize_setup_keywords', group='setuptools.finalize_distribution_options')
EntryPoint(name='parent_finalize', value='setuptools.dist:_Distribution.finalize_options', group='setuptools.finalize_distribution_options')
EntryPoint(name='cffi_modules', value='cffi.setuptools_ext:cffi_modules', group='distutils.setup_keywords')
EntryPoint(name='conda-content-trust', value='conda_content_trust.cli:cli', group='console_scripts')
EntryPoint(name='pip', value='pip._internal.cli.main:main', group='console_scripts')
EntryPoint(name='pip3', value='pip._internal.cli.main:main', group='console_scripts')
EntryPoint(name='pip3.9', value='pip._internal.cli.main:main', group='console_scripts')
EntryPoint(name='conda-pack', value='conda_pack.cli:main', group='console_scripts')
EntryPoint(name='conda-libmamba-solver', value='conda_libmamba_solver.plugin', group='conda')
```

Reply:
Oooh, interesting, why is the solver loaded twice, though? In doubt, I'd suggest to ignore the second run-through again and undo this part

Reply:
I've opened https://github.com/conda/conda/pull/13118 for that